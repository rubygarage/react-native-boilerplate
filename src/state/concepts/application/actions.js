import * as types from './types';

export const setOnboardingCompletion = (isCompleted) => ({
  type: types.SET_ONBOARDING_COMPLETION,
  isCompleted,
});

export const startListenConnectionState = () => ({
  type: types.START_LISTEN_CONNECTION_STATE,
});

export const stopListenConnectionState = () => ({
  type: types.STOP_LISTEN_CONNECTION_STATE,
});

export const setConnectionState = (isConnected) => ({
  type: types.SET_CONNECTION_STATE,
  isConnected,
});
