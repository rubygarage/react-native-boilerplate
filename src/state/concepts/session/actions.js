import * as types from './types';

export const setTokens = (tokens) => ({
  type: types.SET_TOKENS,
  tokens,
});

export const removeTokens = () => ({
  type: types.SET_TOKENS,
  tokens: null,
});

export const setUserId = (id) => ({
  type: types.SET_USER_ID,
  id,
});
