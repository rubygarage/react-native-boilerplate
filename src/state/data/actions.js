import * as types from './types';

export const dataApiRequest = ({ endpoint }) => ({
  type: types.DATA_API_REQUEST,
  endpoint,
});

export const dataApiSuccess = ({ endpoint, response }) => ({
  type: types.DATA_API_SUCCESS,
  endpoint,
  response,
});

export const dataApiFailure = ({ endpoint }) => ({
  type: types.DATA_API_FAILURE,
  endpoint,
});

export const dataDelete = ({ kind, ids }) => ({
  type: types.DATA_DELETE,
  kind,
  ids,
});
