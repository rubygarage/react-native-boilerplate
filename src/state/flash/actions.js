import { ALERT_TYPES } from 'constants/alert';
import * as types from './types';

export const showFlash = ({
  id,
  lifetime,
  priority,
  isSingleShot = false,
  flashProps,
}) => ({
  type: types.SHOW_FLASH,
  flash: {
    id,
    lifetime,
    priority,
    isSingleShot,
    flashProps,
  },
});

export const showErrorFlash = (action) => ({
  type: types.SHOW_FLASH,
  flash: {
    isSingleShot: false,
    flashProps: {
      title: { id: 'shared.oops' },
      type: ALERT_TYPES.error,
      action,
      actionTextId: 'shared.tryAgain',
      isCloseable: true,
    },
  },
});

export const addFlash = (flash) => ({
  type: types.ADD_FLASH,
  flash,
});

export const hideFlash = (id) => ({
  type: types.HIDE_FLASH,
  id,
});

export const removeFlash = (id, isSingleShot) => ({
  type: types.REMOVE_FLASH,
  id,
  isSingleShot,
});
