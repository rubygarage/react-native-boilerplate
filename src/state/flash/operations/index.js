import hideFlash from './hideFlash';
import showFlash from './showFlash';

export default [
  hideFlash, showFlash,
];
