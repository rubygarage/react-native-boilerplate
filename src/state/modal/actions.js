import * as types from './types';

export const showModal = ({ modal, modalProps }) => ({
  type: types.SHOW_MODAL,
  modal,
  modalProps,
});

export const hideModal = () => ({
  type: types.HIDE_MODAL,
});
