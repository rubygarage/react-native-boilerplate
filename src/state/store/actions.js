import * as types from './types';

// eslint-disable-next-line import/prefer-default-export
export const resetStore = () => ({
  type: types.RESET_STORE,
});
