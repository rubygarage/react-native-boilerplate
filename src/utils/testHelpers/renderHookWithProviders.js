/* eslint-disable import/no-extraneous-dependencies */
import { renderHook } from '@testing-library/react-hooks';
import { ThemeProvider } from 'styled-components';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import mockedTheme from 'utils/testHelpers/mockedTheme';

const store = configureStore()({});
store.dispatch = jest.fn();

const renderHookWithProviders = (hook) => {
  const wrapper = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={mockedTheme}>
        {children}
      </ThemeProvider>
    </Provider>
  );

  return renderHook(hook, { wrapper });
};

export default renderHookWithProviders;
