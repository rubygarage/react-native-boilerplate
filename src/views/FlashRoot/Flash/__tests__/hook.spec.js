import { act } from '@testing-library/react-hooks';

import { hideFlash } from 'state/flash/actions';
import renderHookWithProviders from 'utils/testHelpers/renderHookWithProviders';

import useContainer from '../hook';

jest.useFakeTimers();

const mockedDispatch = jest.fn();

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockedDispatch,
}));

describe('doAnimationHook', () => {
  const props = {
    id: 'some',
    flashProps: { action: { type: 'TEST' } },
    doAnimation: jest.fn(),
    lifetime: null,
  };

  let { result, unmount } = renderHookWithProviders(() => useContainer(props));

  it('matches snapshot', () => {
    expect(result.current).toMatchSnapshot();
  });

  it('checks onHideFlash method', () => {
    act(() => {
      result.current.onHideFlash();
    });

    expect(mockedDispatch).toHaveBeenCalledWith(hideFlash(props.id));
    expect(props.doAnimation).toHaveBeenCalled();
  });

  it('checks onActionClick method', () => {
    mockedDispatch.mockClear();

    act(() => {
      result.current.onActionClick();
    });

    expect(mockedDispatch).toHaveBeenCalledTimes(2);
    expect(mockedDispatch).toHaveBeenNthCalledWith(1, hideFlash(props.id));
    expect(mockedDispatch).toHaveBeenNthCalledWith(2, { type: 'TEST' });
  });

  describe('Lifecycle', () => {
    it('mounting - should not call setInterval', () => {
      expect(setInterval).not.toHaveBeenCalledWith();
    });

    it('mounting - should call setInterval', () => {
      result = renderHookWithProviders(() => useContainer({ ...props, lifetime: 1000 })).result;

      expect(setInterval).toHaveBeenCalledWith(result.current.onHideFlash, 1000);

      expect(clearInterval).toHaveBeenCalled();
    });

    it('checks cleanup callbacks', () => {
      ({ result, unmount } = renderHookWithProviders(
        () => useContainer({ ...props, lifetime: 1000 }),
      ));

      unmount();

      expect(clearInterval).toHaveBeenCalled();
    });
  });
});
