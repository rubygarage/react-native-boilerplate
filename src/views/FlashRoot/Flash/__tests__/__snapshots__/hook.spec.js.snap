// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doAnimationHook matches snapshot 1`] = `
Object {
  "onActionClick": [Function],
  "onHideFlash": [Function],
}
`;
